src/SwaggerProvider.DesignTime/Provider.SwaggerClient.fs

```diff
@@ -61,27 +63,40 @@ type public SwaggerTypeProvider(cfg : TypeProviderConfig) as this =
                 let ignoreControllerPrefix = unbox<bool>  args.[3]
                 let preferNullable = unbox<bool>  args.[4]
                 let preferAsync = unbox<bool>  args.[5]
+                let resolveReferences = unbox<bool>  args.[6]
 
                 let cacheKey =
-                    (schemaPathRaw, headersStr, ignoreOperationId, ignoreControllerPrefix, preferNullable, preferAsync)
+                    (schemaPathRaw, headersStr, ignoreOperationId, ignoreControllerPrefix, preferNullable, preferAsync, resolveReferences)
                     |> sprintf "%A"
 
                 let addCache() =
                   lazy
                     let schemaData =
                         SwaggerProvider.Internal.SchemaReader.readSchemaPath headersStr schemaPathRaw
                         |> Async.RunSynchronously
-                    let schema = SwaggerParser.parseSchema schemaData
+                    let mainSchema = SwaggerParser.parseSchema resolveReferences schemaData
 
-                    let defCompiler = DefinitionCompiler(schema, preferNullable)
-                    let opCompiler = OperationCompiler(schema, defCompiler, ignoreControllerPrefix, ignoreOperationId, preferAsync)
-                    opCompiler.CompileProvidedClients(defCompiler.Namespace)
-                    let tys = defCompiler.Namespace.GetProvidedTypes()
+                    let compile schema = 
+                        let defCompiler = DefinitionCompiler(schema, preferNullable)
+                        let opCompiler = OperationCompiler(schema, defCompiler, ignoreControllerPrefix, ignoreOperationId, preferAsync)
+                        opCompiler.CompileProvidedClients(defCompiler.Namespace)
+                        defCompiler.Namespace.GetProvidedTypes()
+
+                    let tys = compile mainSchema
+
+                    let rec fetchExternalSchemaTypes (schema:Schema.SwaggerObject) (collected:ProvidedTypeDefinition list) =
+                        if Array.isEmpty schema.SwaggerObjects then collected
+                        else schema.SwaggerObjects |> Array.toList |> List.map(fun eSchema ->
+                                  compile eSchema |> fetchExternalSchemaTypes eSchema) |> List.concat
+
+                    let externalTys = fetchExternalSchemaTypes mainSchema []
```

Comment:
Can  we move this line inside `if resolveReferences then` and do not fetch external types when user does not request it.